### PR TITLE
[Refactor] Broadcast compact RoPE tables in MAGI-2

### PR DESCRIPTION
**Pull request by yeahdongcn**
## Summary

Avoid duplicated cos/sin tensors in MAGI-2's non-interleaved Torch RoPE helper.

Use a view of the two rotary halves and broadcast compact tables over that axis. Keep the existing `rotate_half` calculation, negation-before-multiplication, multiply/add order and PyTorch dtype promotion. Interleaved inputs and unequal table shapes retain their original path.

This is a portable, precision-preserving extraction of the RoPE work in reference #7156, not a copy of its MUSA-only table downcast. For example, BF16 inputs with FP32 tables still produce FP32 output, including promotion of the untouched tail. Current MAGI-2 normalizes Q/K to FP32 before RoPE and casts afterward; that model boundary is unchanged.

## Scope

Two files only: 11 implementation lines and focused tests. No new kernel, backend dispatch, environment variable, registry entry, cache, sampler or attention-provider change. Shared `RotaryEmbedding` and its other backend/full-dimension contracts are untouched.

The eager operation test counts two `torch.cat` calls instead of four: rotation and output concatenation remain; the two duplicated-table concatenations disappear. This is a structural observation, not an E2E speed or peak-memory measurement.

## Validation

Tested commit: `bfb11303f0a7266ad6187316219724bed527eee0`, one signed-off commit based on upstream `d59d1664858956976a12797a8613bd4b1d803097`.

- Targeted pre-commit passed, including Ruff, mypy, test markers and CUDA API checks.
- **63 CPU tests passed**, 4 MUSA deselected: exact original-formula output parity, mixed dtypes, both rotary layouts, noncontiguous/batched/empty inputs, zero rotary dimension, partial tails, unusual broadcast/error behavior, input preservation, signed zeros/nonfinite values, integer negation and gradients; plus existing native model and shared RoPE broadcast tests.
- The two-layer native MAGI-2 test compares complete output against the pre-refactor formula and observes four Q/K RoPE calls (`rtol=atol=0`).
- A Dynamo test asserts that graphs are actually produced across repeated shapes. Its backend executes the FX graph eagerly; this is **not** Inductor or graph-capture validation.
- **4 MUSA tests passed** on one 48-SM MTT S5000 (driver `5.2.0-server`). FP32/FP32, BF16/BF16, BF16/FP32 and FP16/FP32 input/table combinations compare to the original formula with **rtol=atol=0**. They exercise both layouts, noncontiguous GPU inputs, empty/zero rotary cases, complete D128 rotation and partial 96/128 rotation.

Stack: torch/torch_musa `2.11.0.post1+musa5.2.0`, torchada `0.1.83`, vLLM `0.28.0`, vllm-musa `0.1.28`. Exact source installed editable without changing dependencies; the MUSA entrypoint imports torchada first.

```bash
python -m pytest -q -o addopts= -m cpu \
  tests/diffusion/models/magi2/test_pairwise_rope.py \
  tests/diffusion/models/magi2/test_native_preview.py \
  tests/diffusion/layers/test_rope_broadcast.py
```

For MUSA, run the new test file with `-m musa` from a torchada-first entrypoint with one visible GPU.

**Not run:** CUDA hardware, GPU gradients, Inductor/capture, full MUSA MAGI-2 checkpoint/video, peak-memory or performance benchmarks. This PR does not assert a DiT-step latency improvement or resolve the separate FA3 consumer's numerical gates. It remains Draft.

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
This PR touches **tests/diffusion/, vllm_omni/diffusion/** (2 files). Based on CODEOWNERS coverage of the changed files, the most-related reviewers appear to be:

> @fhfuih @Bounty-hunter @wtomin

Could one of you take a look when you get a chance? Thanks!
